content/en/borealis/Reference/ref-deployment-file.md

```diff
@@ -343,4 +357,206 @@ steps:
+      lowerLimit: <integer> # If the metric goes below this value, the automated analysis fails.
+      queryTemplate: >-
+        <some-metrics-query>
+    - name: <queryName>
+      upperLimit: <integer>  # If the metric exceeds this value, the automated analysis fails.
+      lowerLimit: <integer> # If the metric goes below this value, the automated analysis fails.
+      context:
+        - <variableName>: <Variable>
+        - <variableName>: <Variable>
+      queryTemplate: >-
+        <some-metrics-query>
+```
+
+### `analysis.defaultMetricProviderName`
+
+The name that you assigned to a metrics provider in the **Configuration UI**.
+
+### `analysis.queries`
+
+This block is how you define the queries that you want to run.
+
+#### `analysis.queries.name`
+
+Used in `analysis` steps to specify the query that you want to use for the step. Specifically it's used for the list in `steps.analysis.queries`.
+
+Provide a unique and descriptive name for the query, such as `containerCPUSeconds` or `avgMemoryUsage`.
+
+#### `analysis.queries.upperLimit`
+
+The upper limit for the query. If the analysis returns a value that is above this range, the deployment is considered a failure, and a rollback is triggered. The rollback can happen either manually or automatically depending on how you configured `strategies.<strategyName>.<strategy>.steps.analysis.rollBackMode`.
+
+If the query returns a value that falls within the range between the `upperLimit` and `lowerLimit` after all the runs of the query complete, the query is considered a success.
+
+#### `analysis.queries.lowerLimit`
+
+The lower limit for the query. If the analysis returns a value that is below this range, the deployment is considered a failure, and a rollback is triggered. The rollback can happen either manually or automatically depending on how you configured `strategies.<strategyName>.<strategy>.steps.analysis.rollBackMode`.
+
+If the query returns a value that falls within the range between the `upperLimit` and `lowerLimit` after all the runs of the query, the query is considered a success.
+
+#### `analysis.queries.queryTemplate`
+
+```yaml
+analysis: # Define queries and thresholds used for automated analysis
+  queries:
+    - name: <queryName>
+      upperLimit: <integer> # If the metric exceeds this value, the automated analysis fails.
+      lowerLimit: <integer> # If the metric goes below this value, the automated analysis fails.
+      queryTemplate: >-
+        <some-metrics-query>
+    - name: <queryName>
+      upperLimit: <integer>  # If the metric exceeds this value, the automated analysis fails.
+      lowerLimit: <integer> # If the metric goes below this value, the automated analysis fails.
+      queryTemplate: >-
+        <some-metrics-query>
+```
+
+The query you want to run. Use the **Retrospective Analysis** UI to build and test queries before including them in your deploy file.
```

Comment:
it will be great to link to the Retrospective Analysis steps here. in case someone lands to this page directly from google or some other way.